packages/flutter_test/lib/src/controller.dart

```diff
@@ -217,6 +316,309 @@ class SemanticsController {
+  /// by `finder`
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  /// [SemanticsAction.scrollUp]
+  void scrollUp(Finder finder) {
+    performAction(findScrollable(finder), SemanticsAction.scrollUp);
+  }
+
+  /// Performs a [SemanticsAction.scrollDown] action on the [SemanticsNode]
+  /// found by `finder`
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  /// [SemanticsAction.scrollDown]
+  void scrollDown(Finder finder) {
+    performAction(findScrollable(finder), SemanticsAction.scrollDown);
+  }
+
+  /// Performs a [SemanticsAction.increase] action on the [SemanticsNode]
+  /// found by `finder`
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  /// [SemanticsAction.increase]
+  void increase(Finder finder) {
+    final SemanticsNode node = findWithAction(finder, SemanticsAction.increase);
+    performAction(node, SemanticsAction.increase);
+  }
+
+  /// Performs a [SemanticsAction.decrease] action on the [SemanticsNode]
+  /// found by `finder`
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  /// [SemanticsAction.decrease]
+  void decrease(Finder finder) {
+    final SemanticsNode node = findWithAction(finder, SemanticsAction.decrease);
+    performAction(node, SemanticsAction.decrease);
+  }
+
+  /// Performs a [SemanticsAction.showOnScreen] action on the [SemanticsNode]
+  /// found by `finder`
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  /// [SemanticsAction.showOnScreen]
+  void showOnScreen(Finder finder) {
+    performAction(
+      find(finder),
+      SemanticsAction.showOnScreen,
+      checkForAction: false,
+    );
+  }
+
+  /// Performs a [SemanticsAction.moveCursorForwardByCharacter] action on the
+  /// [SemanticsNode] found by `finder`
+  ///
+  /// If `shouldModifySelection` is true, then the cursor will begin or extend
+  /// a selection.
+  ///
+  /// Throws a [StateError] if:
+  /// * The given `finder` returns zero or more than one result
+  /// * The [SemanticsNode] found with `finder` does not support
+  /// [SemanticsAction.moveCursorForwardByCharacter]
+  void moveCursorForwardByCharacter(Finder finder, [bool shouldModifySelection = false]) {
+    final SemanticsNode node = findWithAction(finder, SemanticsAction.moveCursorForwardByCharacter);
+    performAction(
+      node,
+      SemanticsAction.moveCursorForwardByCharacter,
+      args: shouldModifySelection);
```

Comment:
nit: here and everywhere else: format this like in showOnScreen.

Reply:
Done!